tests/bsim/bluetooth/host/l2cap/multilink_peripheral/src/central.c

```diff
@@ -0,0 +1,164 @@
+/*
+ * Copyright (c) 2024 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/kernel.h>
+#include <zephyr/bluetooth/bluetooth.h>
+#include <zephyr/bluetooth/conn.h>
+#include <zephyr/bluetooth/att.h>
+#include <zephyr/bluetooth/l2cap.h>
+#include <zephyr/logging/log.h>
+
+#include "testlib/scan.h"
+#include "testlib/conn.h"
+
+#include "babblekit/flags.h"
+#include "babblekit/sync.h"
+#include "babblekit/testcase.h"
+
+/* local includes */
+#include "data.h"
+
+LOG_MODULE_REGISTER(central, CONFIG_APP_LOG_LEVEL);
+
+static int total_rx;
+static struct bt_l2cap_le_chan le_chan;
+
+void sent_cb(struct bt_l2cap_chan *chan)
+{
+	TEST_FAIL("Tester should not send data");
+}
+
+int recv_cb(struct bt_l2cap_chan *chan, struct net_buf *buf)
+{
+	LOG_DBG("received %d bytes", buf->len);
+	total_rx++;
+
+	return 0;
+}
+
+void l2cap_chan_connected_cb(struct bt_l2cap_chan *chan)
+{
+	LOG_DBG("%p", chan);
+}
+
+void l2cap_chan_disconnected_cb(struct bt_l2cap_chan *chan)
+{
+	LOG_DBG("%p", chan);
+}
+
+static struct bt_l2cap_chan_ops ops = {
+	.connected = l2cap_chan_connected_cb,
+	.disconnected = l2cap_chan_disconnected_cb,
+	.recv = recv_cb,
+	.sent = sent_cb,
+};
+
+int server_accept_cb(struct bt_conn *conn, struct bt_l2cap_server *server,
+		     struct bt_l2cap_chan **chan)
+{
+	memset(&le_chan, 0, sizeof(le_chan));
+	le_chan.chan.ops = &ops;
+	*chan = &le_chan.chan;
+
+	return 0;
+}
+
+static struct bt_l2cap_server test_l2cap_server = {.accept = server_accept_cb};
+
+static int l2cap_server_register(bt_security_t sec_level)
+{
+	test_l2cap_server.psm = 0;
+	test_l2cap_server.sec_level = sec_level;
+
+	int err = bt_l2cap_server_register(&test_l2cap_server);
+
+	TEST_ASSERT(err == 0, "Failed to register l2cap server (err %d)", err);
+
+	return test_l2cap_server.psm;
+}
+
+static void acl_connected(struct bt_conn *conn, uint8_t err)
+{
+	char addr[BT_ADDR_LE_STR_LEN];
+
+	bt_addr_le_to_str(bt_conn_get_dst(conn), addr, sizeof(addr));
+
+	if (err) {
+		LOG_ERR("Failed to connect to %s (0x%02x)", addr, err);
+		return;
+	}
+
+	LOG_DBG("Connected to %s", addr);
+}
+
+static void acl_disconnected(struct bt_conn *conn, uint8_t reason)
+{
+	char addr[BT_ADDR_LE_STR_LEN];
+
+	bt_addr_le_to_str(bt_conn_get_dst(conn), addr, sizeof(addr));
```

Comment:
Unrelated to this PR, but I just recently learned that `errno` is thread-local in C11 and newer. I wonder if we can somehow apply that same approach to values such as `addr` here, so that we can omit the buffer and simply do `LOG_DBG("%s", bt_addr_le_to_str(..));` at some point?

Reply:
iunderstandsomeofthesewords.jpg 😅
What do you mean exactly?

Reply:
oh yeah that sounds better. didn't know we had proper thread-local integration. I only knew about this https://docs.zephyrproject.org/latest/kernel/services/threads/index.html#thread-custom-data

Reply:
I actually don't know if https://en.cppreference.com/w/c/thread/thread_local is really supporting in Zephyr. I don't see any uses of it, although I do see that we have `CONFIG_THREAD_LOCAL_STORAGE`, so unsure about the state of it. 

But it would definitely be a way to improve several of these type of functions